### ISSUE TITLE
Allow to parse 1 to boolean for SupplyParameterFromQuery attribute in Blazor app

### ISSUE THREAD

**Issue by MarvinKlein1508** [dotnet/aspnetcore#50726]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I have a parameter which is defined as boolean
```csharp
[Parameter, SupplyParameterFromQuery]
public bool Copy { get; set; }
```

However I get this exception when supplying 1 for the parameter 
```
System.InvalidOperationException: "Cannot parse the value '1' as type 'System.Boolean' for 'Copy'."
```

### Describe the solution you'd like

It is fairly common to use 1 for boolean true and 0 for false. So IMO it should work out of the box to supply 0 and 1 for boolean parameters.

**Comment by javiercn** [comment]
@MarvinKlein1508 thanks for contacting us.

I don't think anything on the stack supports this behavior (not Blazor and not MVC) and this is the first time we've received such ask.

**Comment by javiercn** [comment]
@MarvinKlein1508 sure, it's a 13-year-old issue.

If this hasn't driven more asks it's likely that it can easily be accomplish in other ways and not a priority and that the view count doesn't accurately reflect the actual interest. It's very easy to land on a page to realize it's not what you are looking for.

**Comment by MarvinKlein1508** [comment]
@javiercn Yes but you can see on the amount of upvotes that many people were interested in this in some way or another. 

I'Ve just thought about it. Isn't this a usecase to override the `SetParametersAsync` method in a blazor app?

**Comment by gragra33** [comment]
@MarvinKlein1508 `QueryString` is just that, string key and value. So, properties have getters and setters, so you can convert the `string?` value to `boolean?` and back. Something like:
```cs
private bool? _copy;

[Parameter, SupplyParameterFromQuery]
public string? Copy
{
    get => _copy is null ? "" : _copy == true ? "1" : "0";
    set => _copy = value is null ? null : value == "1";
}
```

**Comment by MarvinKlein1508** [comment]
@gragra33 yes but I don't want to parse my Copy property then evertime I use it.

**Comment by gragra33** [comment]
@MarvinKlein1508 There are no other options or converters. `SupplyParameterFromQuery` is `sealed`, so you can't write a wrapper to hide the implementation. I have not tried a facade, but I think that will not work.

Either store as 'string' and apply logic when required or as above.

**Comment by schmellow** [comment]
It probably would help if the behavior was documented somehow?
So SupplyParameterFromQuery at least works automagically on numbers and dates (but i assume only for ISO format for the latter), but can blow up on booleans. 
Simple example: ```<form>``` (get method by default) + checkbox. Checkbox value gets translated as "?checkbox=on" on submit, which does not parse. But then if you go interactive with onsubmit handler and just bind that checkbox to property directly without the attribute, it just works. Feels inconsistent and confusing

**Comment by MarvinKlein1508** [comment]
@schmellow good point. This could also be interesting for SSR forms which submits forms via GET.

**Comment by javiercn** [comment]
Duping against https://github.com/dotnet/aspnetcore/issues/49546 which will be used for tracking purposes
